javascript/javascript1/week1/homework_soheib/goodboy_oldboy.js

```diff
@@ -0,0 +1,12 @@
+let dogYearOfBirth = 2014;
+let dogYearFuture = 2025;
+let dogYear = 7;
+let shouldShowResultInDogYears = false;
+
+if (shouldShowResultInDogYears === true) {
+    let result = (dogYearFuture - dogYearOfBirth) * dogYear;
+    console.log(`Your  dog will be ${result} human years old in ${dogYearFuture}`)
+} else if(shouldShowResultInDogYears === false){
+    result = dogYearFuture - dogYearOfBirth;
```

Comment:
have you tested this? This should throw an error as result has not been initialized as a variable. Remember that the code in line 7 and 8 is not run if shouldShowResultInDogYears is not true :)

Reply:
Oh I missed it, but for some reason it wasn't throwing an error, because I tested it on the browser and it wasn't running in strict mode. I asked about it before, and I was told that strict mode is by default now. However according to chatGPT that's not the case because it would break legacy code. 
I corrected it and tried to make the code shorter but still human-readable. 
Thanks a lot :D